lms/templates/ccx/underscore/form.underscore

```diff
@@ -0,0 +1,181 @@
+<div class="field">
+  <label for="ccx_chapter" class="form-label"><b><%- gettext('Section') %></b></label>
+  <select name="chapter" id="ccx_chapter" >
+    <option value="none"><%- gettext("Select a chapter...") %></option>
+    <% _.each(chapters, function(chapter) { %>
+        <% if (selectedChapter && selectedChapter.location === chapter.location) { %>
+            <option value="<%= HtmlUtils.ensureHtml(chapter.location) %>" selected>
+              <%= HtmlUtils.ensureHtml(chapter.display_name) %>
```

Comment:
It looks like you are confused about the proper usage of HtmlUtils.ensureHtml. It should only be used when the string is expected to be HTML-- for instance, you have constructed a snippet of HTML programmatically, or you are passing in the contents of a template (this is typically pretty rare). The reason you use it is because you don't want actual HTML to be escaped. But in all other cases, we do want text to be escaped. The chapter display_name here is set by the author, and it could nefariously include HTML or a script tag-- you do want to escape the display_name.

<%- chapter.display_name %>

Go through all usages of HtmlUtils in this PR and make sure you are only marking things as HTML if they actually are HTML-- I suspect that most of the usages are incorrect.

More information (from http://edx-developer-guide.readthedocs.io/en/latest/conventions/safe_templates.html):

> The following HtmlUtils functions all make use of HtmlUtils.HtmlSnippet. An HTML snippet is used to communicate to other functions that the string it represents contains HTML that has been safely escaped as necessary.
> 
> The HtmlUtils.ensureHtml() function will ensure you have properly escaped HTML by HTML-escaping any plain text string, or simply returning any HTML snippet provided to it.